platform-sdk/swirlds-merkledb/src/main/java/com/swirlds/merkledb/collections/AbstractLongList.java

```diff
@@ -781,20 +784,32 @@ private void checkValue(final long value) {
         }
     }
 
-    /** {@inheritDoc} */
+    /**
+     * {@inheritDoc}
+     *
+     * <p>This long list implementation checks the condition (if not null) before every list
+     * item is processed.
+     */
     @Override
-    public <T extends Throwable> void forEach(final LongAction<T> action) throws InterruptedException, T {
+    public <T extends Throwable> boolean forEach(
+            @NonNull final LongAction<T> action, @Nullable final BooleanSupplier cond) throws InterruptedException, T {
         final long max = maxValidIndex.get();
         if (max < 0) {
             // Empty list, nothing to do
-            return;
+            return true;
         }
-        for (long i = minValidIndex.get(); i <= max; i++) {
+        Objects.requireNonNull(action);
+        long i = minValidIndex.get();
+        for (; i <= max; i++) {
+            if ((cond != null) && !cond.getAsBoolean()) {
+                break;
+            }
```

Comment:
```suggestion
        BooleanSupplier condition = cond != null ? cond : () -> true;
        for (; i <= max && condition.getAsBoolean(); i++) {
```

Reply:
I wonder if a new lambda would be faster or slower than a null check...

Reply:
It's a `null` lambda in tests only.